internal/engine/docker_pruner.go

```diff
@@ -0,0 +1,76 @@
+package engine
```

Comment:
for future subscribers, can you put them in a subpackage of engine? been trying to reduce the circular dependencies in this code